### PR TITLE
Add services commands

### PR DESCRIPTION
**Pull request by avichani**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -196,12 +212,6 @@
 			"type": "object",
 			"title": "CloudFoundryTools",
 			"properties": {
```
Comment (alex-gilin) [inline package.json:214]: Seems like you branch is not updated. Sync your branch first

File: tests/extension.spec.ts [hunk 2]
```diff
@@ -53,6 +53,8 @@ describe('Extension unit test', () => {
         "cf.deploy-service.api": cfViewCommands.cmdDeployServiceAPI,
         "cf.services.binding.state" : dependencyHandler.getBindState,
         "cf.services.get-space-services": cfViewCommands.cmdGetSpaceServices,
+	    "cf.services.get-ups-services": cfViewCommands.cmdGetUpsServiceInstances,
+        "cf.services.get-services": cfViewCommands.cmdGetServiceInstances,
```
Comment (alex-gilin) [inline tests/extension.spec.ts:57]: format text

File: tests/utils.spec.ts [hunk 3]
```diff
@@ -358,7 +358,7 @@ describe('utils unit tests', () => {
             const expectedQuery = _.cloneDeep(opts.query);
             expectedQuery.filters = _.drop(expectedQuery.filters);
             sandbox.stub(cfLocal, 'cfGetServiceInstances').resolves(services);
-            sandbox.stub(cfLocal, 'cfGetUpsInstances').withArgs(expectedQuery).resolves(ups);
+            sandbox.stub(cfLocal, 'cfGetUpsInstances').resolves(ups);
```
Comment (alex-gilin) [inline tests/utils.spec.ts:361]: you remove expected args verification

**Review by alex-gilin — CHANGES_REQUESTED** [review]
Seems like your branch is not updated vs master

File: src/cfViewCommands.ts [hunk 4]
```diff
@@ -204,18 +215,21 @@ export async function cmdDeleteTarget(item: unknown): Promise<void> {
...
 }
 
-async function composePlansGuidListForQuery(serviceInfos: ServiceTypeInfo[], planList: PlanInfo[]): Promise<string[]> {
-    let plans: string[] = [];
-    if(_.get(serviceInfos, ['0', 'plan'])) {
-        plans = _.map(_.filter(planList, ['label', serviceInfos[0].plan]), 'guid');
-    }
-    if (!_.size(plans) && _.get(serviceInfos, '[0].name')) {
-        plans = await getServicePlansGuidList(_.get(serviceInfos, '[0].name'));
+export async function cmdGetUpsServiceInstances(options?: UpsServiceQueryOprions, progressTitle?: string): Promise<ServiceInstanceInfo[]> {
```
Comment (alex-gilin) [inline src/cfViewCommands.ts:221]: what is a purpose of this wrapper function ? seems like it is redundant

File: src/utils.ts [hunk 5]
```diff
@@ -211,18 +225,27 @@ export function composeFilterPattern(value: string): string {
...
+	const upsServices = await cfGetUpsInstances(query);
+	const pattern = (_.size(upsServices) && filterCredTag) ? composeFilterPattern(filterCredTag) : undefined;
+	const ups2Show  = pattern ? upsServices.filter(service => {
+	  return _.find(service.credentials?.tags, (tag) => new RegExp(pattern).test(tag));
+	}) : upsServices;
+	return ups2Show;
+}
+			
+export async function getUpsServiceInstances(options?: UpsServiceQueryOprions): Promise<ServiceInstanceInfo[]> {
+	return doGetUpsServiceInstances(undefined, options?.credentials?.tag);
```
Comment (alex-gilin) [inline src/utils.ts:238]:
```suggestion
	return doGetUpsServiceInstances(options?.query, options?.credentials?.tag);
```

File: tests/commands.spec.ts [hunk 6]
```diff
@@ -779,6 +779,46 @@ describe("commands unit tests", () => {
...
+            const result = await commands.getServiceInstances();
+            expect(result).to.have.lengthOf(1);
+        });
+
+        it("on any error", async () => {
+            const errorMessage = "test error";
+            vscodeWindowMock.expects("withProgress").withArgs({ location: nsVsMock.testVscode.ProgressLocation.Notification, title: messages.loading_services, cancellable: true }).rejects(new Error(errorMessage));
+            try {
+                await commands.getServiceInstances();
+                expect(true).to.be.equal(false);
```
Comment (alex-gilin) [inline tests/commands.spec.ts:795]:
```suggestion
                fail("test should fail here");
```

File: tests/commands.spec.ts [hunk 7]
```diff
@@ -779,6 +779,46 @@ describe("commands unit tests", () => {
...
+        });
+    });
+
+    describe("getUserProvidedServiceInstances", () => {
+
+        it("on success", async () => {
+            vscodeWindowMock.expects("withProgress").withArgs({ location: nsVsMock.testVscode.ProgressLocation.Notification, title: messages.loading_services, cancellable: true }).resolves([{}]);
+            const result = await commands.getUserProvidedServiceInstances();
+            expect(result).to.have.lengthOf(1);
+        });
```
Comment (alex-gilin) [inline tests/commands.spec.ts:808]: what exactly does the test check?

File: tests/commands.spec.ts [hunk 8]
```diff
@@ -779,6 +779,46 @@ describe("commands unit tests", () => {
...
+
+        it("on any error", async () => {
+            const errorMessage = "test error";
+            vscodeWindowMock.expects("withProgress").withArgs({ location: nsVsMock.testVscode.ProgressLocation.Notification, title: messages.loading_services, cancellable: true }).rejects(new Error(errorMessage));
+            try {
+                await commands.getUserProvidedServiceInstances();
+                expect(true).to.be.equal(false);
```
Comment (alex-gilin) [inline tests/commands.spec.ts:815]:
```suggestion
                fail("should fail here");
```

**Review by alex-gilin — CHANGES_REQUESTED** [review]

File: tests/commands.spec.ts (hunk 6 above)
Comment (avichani) [reply tests/commands.spec.ts:795]: fix, note that there is a better way to handle exception but I didn't want to change all the test 
read about "chai-as-promised" package

File: tests/commands.spec.ts (hunk 8 above)
Comment (avichani) [reply tests/commands.spec.ts:815]: done

File: tests/commands.spec.ts (hunk 7 above)
Comment (avichani) [reply tests/commands.spec.ts:808]: Let review it

**Review by alex-gilin — CHANGES_REQUESTED** [review]

**Review by alex-gilin — APPROVED** [review]

**Pull request merged**